src/colorizer/utils/data_utils.ts

```diff
@@ -85,3 +85,38 @@ export function isValueWithinThreshold(value: number, threshold: FeatureThreshol
     return threshold.enabledCategories[value];
   }
 }
+
+/**
+ * Returns a Uint8 array look-up table indexed by object ID, storing whether that object ID is in range of
+ * the given thresholds (=1) or not (=0).
+ * @param {Dataset} dataset A valid Dataset object.
+ * @param {FeatureThreshold[]} thresholds Array of feature thresholds, which match agaisnt the feature name and unit.
+ * If a feature name cannot be found in the dataset, it will be ignored.
+ * @returns A Uint8Array, with a length equal to the number of objects in the dataset.
+ * For each object ID `i`, `inRangeIds[i]` will be 1 if the object is in range of the thresholds
+ * and 0 if it is not.
+ */
+export function getInRangeLUT(dataset: Dataset, thresholds: FeatureThreshold[]): Uint8Array {
+  // TODO: Optimize memory by using a true boolean array?
```

Comment:
Probably not needed but if this was ever a concern, could also pack 8 boolean bits into each uint8!